components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/api/APIMgtAdminService.java

```diff
@@ -161,4 +162,19 @@
      * @throws APIManagementException if failed to retrieve resources
      */
     List<UriTemplate> getAllResourcesForApi(String apiContext, String apiVersion) throws APIManagementException;
+
+    /**
+     * Get a list of APIs with given gateway labels and status
+     *@param gatewayLabels A list of gateway labels
+     * @param status Lifecycle status
+     * @throws APIManagementException If failed to get API list
+     */
+    List<API> getAPIsByStatus(List<String> gatewayLabels, String status) throws APIManagementException;
```

Comment:
This is for API Manager 3.0 which is under development with C5. We haven't done a major release yet and major development is still on going. Therefore we are continuously doing refactoring and heavy development. After major release we will adhere to the versioning.